Board.java

```diff
@@ -30,33 +28,123 @@ public Board playerMakesMove(char player, int position) {
         return this;
     }
 
+    public String getWinner() {
+        for (String combination : allWinningCombinations()) {
+            if (!combination.contains("x") && !combination.contains("-")) {
+                return "o";
+            }
+            if (!combination.contains("o") && !combination.contains("-")) {
+                return "x";
+            }
+        }
+        return null;
+    }
+
+    public boolean hasWinner() {
+        return getWinner() != null;
+    }
+
+
+    public boolean isGameOver() {
+        return hasWinner() || checkForDraw();
+    }
+
+    public ArrayList<String> allWinningCombinations() {
+        ArrayList<String> combinations = new ArrayList<>();
+        combinations.addAll(getAllRows());
+        combinations.addAll(getAllColumns());
+        return combinations;
+    }
+
+    public ArrayList<String> getAllRows() {
+        ArrayList<String> rows = new ArrayList<>();
+        for (int i=0; i < dimension * dimension; i+=dimension) {
+            String row = "";
+            for (int j = i; j < i + dimension; j++) {
+                row += state.charAt(j);
+            }
+            rows.add(row);
+        }
+        return rows;
+    }
+
+    public ArrayList<String> getAllDiagonals() {
+        ArrayList<String> diagonals = new ArrayList<>();
+        String diagonal = "";
+        for (int j = 0; j < dimension * dimension + 1; j+=dimension + 1) {
```

Comment:
dimension * dimension could be extracted out into a helper method called boardSize(), in order to aid readability